### PR TITLE
fix(relay): get svc name and namespace using service for kubearmor-relay

### PR DESCRIPTION
**Pull request by rajaSahil**
**Purpose of PR?**:

Fixes #
- get svc name and namespace using service for kubearmor-relay

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Review by seswarrajan — APPROVED** [review]
LGTM

File: src/cluster/k8sClientHandler.go [hunk 1]
```diff
@@ -581,26 +581,24 @@ func GetNodesFromK8sClient() (*v1.NodeList, error) {
...
-	var namespace string
 	client := ConnectK8sClient()
 	if client == nil {
 		return ""
 	}
 
 	// get kubearmor-relay pod from k8s api client
-	pods, err := client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+	svc, err := client.CoreV1().Services("").List(context.Background(), metav1.ListOptions{
 		LabelSelector: "kubearmor-app=kubearmor-relay",
```
Comment (vishnusomank) [inline src/cluster/k8sClientHandler.go:591]: are we using this label for kubearmor service?
```
k -n kube-system get svc --show-labels | grep -i kubearmor
kubearmor                                       ClusterIP   10.0.45.5     <none>        32767/TCP       47h   <none>
kubearmor-annotation-manager-metrics-service    ClusterIP   10.0.178.96   <none>        443/TCP         47h   kubearmor-app=kubearmor-annotation-manager
kubearmor-host-policy-manager-metrics-service   ClusterIP   10.0.67.88    <none>        8443/TCP        47h   kubearmor-app=kubearmor-host-policy-manager
kubearmor-policy-manager-metrics-service        ClusterIP   10.0.155.61   <none>        8443/TCP        47h   kubearmor-app=kubearmor-policy-manager
```

Comment (seswarrajan) [reply src/cluster/k8sClientHandler.go:591]: image
I think we are using **kubearmor-app=kubearmor-relay** as the label @vishnusomank

Comment (vishnusomank) [reply src/cluster/k8sClientHandler.go:591]: @seswarrajan this label is for the service pod selector, right? meaning this svc will redirect traffic to the pod with label `kubearmor-app=kubearmor-relay`. I don't think we have attached a label to the kubearmor svc.
```
k -n kube-system get svc kubearmor -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2023-06-21T06:41:49Z"
  name: kubearmor
  namespace: kube-system
  resourceVersion: "8642527"
  uid: 77ab50bf-fec9-4d16-af37-f10fe53bbc6b
spec:
  clusterIP: 10.0.45.5
  clusterIPs:
  - 10.0.45.5
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - port: 32767
    protocol: TCP
    targetPort: 32767
  selector:
    kubearmor-app: kubearmor-relay
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

Comment (Prateeknandle) [reply src/cluster/k8sClientHandler.go:591]: ig `LabelSelector` is used to look for labels under metadata and not `Spec.Selector`. 
`kubearmor-app=kubearmor-relay` is the value of `selector` in `kubearmor` service.
```
apiVersion: v1
kind: Service
metadata:
  name: kubearmor
  namespace: kube-system
spec:
  ports:
  - port: 32767
    protocol: TCP
    targetPort: 32767
  selector:
    kubearmor-app: kubearmor-relay
```
ref : https://github.com/kubearmor/KubeArmor/blob/main/deployments/k3s/kubearmor.yaml#L86
@rajaSahil PTAL at this issue.

Comment (rajaSahil) [reply src/cluster/k8sClientHandler.go:591]: Have raised a PR to add label for service https://github.com/kubearmor/KubeArmor/pull/1275

**Review by Prateeknandle — APPROVED** [review]
LGTM :+1: 
Let's merge this as well.

**Pull request merged**